frontend/packages/dev-console/src/components/edit-deployment/utils/edit-deployment-validation-utils.ts

```diff
@@ -0,0 +1,137 @@
+import * as yup from 'yup';
+import { TFunction } from 'i18next';
```

Comment:
Change to the way things are done for functions. Can you update your functions not to take a `TFunction`.

* Components use `useTranslation` hook
* Functions that use `t` should use `i18n.t` instead as it pulls the global lang state at use
* Don't use `i18n.t` on root-file constants as it only gets executed when the file is imported the first time (cached)

```suggestion
import i18n from '@console/internal/i18n';
```